pkg/virt-operator/install-strategy/strategy.go

```diff
@@ -273,28 +273,42 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
 		}
 	}
 
+	var productName string
+	var productVersion string
+
+	if isValidLabel(config.AdditionalProperties[operatorutil.ProductNameKey]) {
+		productName = config.AdditionalProperties[operatorutil.ProductNameKey]
+	} else {
+		log.Log.Errorf("invalid kubevirt.spec.productName: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
+	}
+	if isValidLabel(config.AdditionalProperties[operatorutil.ProductVersionKey]) {
+		productVersion = config.AdditionalProperties[operatorutil.ProductVersionKey]
+	} else {
+		log.Log.Errorf("invalid kubevirt.spec.productVersion: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
+	}
+
 	strategy.validatingWebhookConfigurations = append(strategy.validatingWebhookConfigurations, components.NewOpertorValidatingWebhookConfiguration(operatorNamespace))
 	strategy.validatingWebhookConfigurations = append(strategy.validatingWebhookConfigurations, components.NewVirtAPIValidatingWebhookConfiguration(config.GetNamespace()))
 	strategy.mutatingWebhookConfigurations = append(strategy.mutatingWebhookConfigurations, components.NewVirtAPIMutatingWebhookConfiguration(config.GetNamespace()))
 
 	strategy.services = append(strategy.services, components.NewPrometheusService(config.GetNamespace()))
 	strategy.services = append(strategy.services, components.NewApiServerService(config.GetNamespace()))
 	strategy.services = append(strategy.services, components.NewOperatorWebhookService(operatorNamespace))
-	apiDeployment, err := components.NewApiServerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetApiVersion(), config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv())
+	apiDeployment, err := components.NewApiServerDeployment(config.GetNamespace(), config.GetImageRegistry(), config.GetImagePrefix(), config.GetApiVersion(), productName, productVersion, config.GetImagePullPolicy(), config.GetVerbosity(), config.GetExtraEnv())
```

Comment:
These method signatures are very long, is there a reason to not pass the whole config object?

Reply:
I'll double check, but I don't see any harm in using the entire object.

Reply:
So it turns out the unit tests here don't have a config object. They basically test the parameters directly. In the interests of not boiling the ocean I think I'm going to defer this.